rust/connlib/tunnel/src/client.rs

```diff
@@ -465,22 +425,23 @@ impl ClientState {
 
         let Some(peer) = peer_by_resource_mut(&self.resources_gateways, &mut self.peers, resource)
         else {
-            // If we don't have a peer but we have an ip for a resource it's then not a DNS resource
-            debug_assert!(self.cidr_resources.longest_match(dest).is_some());
-
-            self.on_connection_intent_to_resource(resource, None, now);
+            self.on_connection_intent_to_resource(resource, &dest, now);
             return None;
         };
 
-        let packet = peer.transform_tun_to_network(packet);
-
         let packet = maybe_mangle_dns_query_to_cidr_resource(
             packet,
             &self.dns_mapping,
             &mut self.mangled_dns_queries,
             now,
         );
 
+        if peer.allowed_ips.longest_match(dest).is_none() {
+            let gateway_id = peer.id();
+            self.send_proxy_ips(&dest, resource, gateway_id);
+            return None;
+        }
```

Comment:
Can you add a comment explaining why we are doing this?

Reply:
yeah, the quick and dirty is that we might have a connection with the desired gateway but we haven't send the ips yet. I'll write a comment when I generating the ipv4/ipv6 proxy ips since this might need a small tweak :)